srcs/dfabuilder.cc

```diff
@@ -0,0 +1,27 @@
+#include <cassert>
+#include "dfabuilder.hpp"
+
+using namespace dfa;
+Dfa::DfaBuilder &Dfa::DfaBuilder::WithState(Dfa::State state, Dfa::Action action){
```

Comment:
Do you want to expose the fact that `Dfa::State` is an integer?  (Your current implementation says yes.)

Reply:
Hide the implementation details.